app/controllers/comments_controller.rb

```diff
@@ -1,21 +1,38 @@
 class CommentsController < ApplicationController
+  load_and_authorize_resource
   before_action :set_user_and_post
 
   def new
-    @comment = Comment.new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
   end
 
   def create
-    @comment = @post.comments.build(comment_params)
-    @comment.user = current_user
+    @user = current_user
+    @post = Post.find(params[:post_id])
+
+    @comment = @post.comments.new(comment_params)
+    @comment.author_id = @user.id # Update this line
```

Comment:
- [ ] When I  create a comment to test and come back to the user show page I get the following error Please make sure to fix it in order to be able to test your authorization
<img width="1178" alt="Screenshot 2023-08-31 at 17 46 15" src="https://github.com/rachaouldbabaali/blog-app/assets/61790587/1715b6a2-13ec-496f-b382-edf8caca7ce4">